src/api/c/indexing_common.cpp

```diff
@@ -0,0 +1,40 @@
+/*******************************************************
+ * Copyright (c) 2018, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <cmath>
+#include <mutex>
+#include <common/err_common.hpp>
+#include <indexing_common.hpp>
+
+af_index_t spanner()
+{
+    static af_index_t s;
+    static std::once_flag flag;
+    std::call_once(flag, []() {
```

Comment:
Kind of complicated way of doing this. Static objects are already thread safe. If we create a factory function that creates these objects, this call_once will go away.

Reply:
This is not about factory based implementation or otherwise. I wanted to do the following. 
```
static const af_index_t s { {.seq=af_span}, true, false };
```
I did and it works fine with GCC, but on MSVC this is failing because using VC compiler we can only assign to the first element of the union `idx` inside the struct `af_index_t`. Although it is supported in C++11, MSVC screwed it up. Therefore, I had to split the declaration and initialization which is where call_once came in.